### PR TITLE
chore: Make analyticsMetadata api public

### PR DESCRIPTION
**Pull request by michaeldowseza**
### Description

This change makes the private `__analyticsMetadata` property public for `instanceIdentifier` and `flowType`. The documenter changes to support the `@analytics` tag can be found in this [documenter PR](https://github.com/cloudscape-design/documenter/pull/39).

Once the documenter PR has been merged/published then the documenter snapshot in this PR will need to be updated.

Related links, issue #, if available: n/a

### How has this been tested?

Tests for the original __analyticsMetadata property have been updated

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/base-component/index.ts [hunk 1]
```diff
@@ -39,9 +39,11 @@ export function getBaseProps(props: BaseComponentProps) {
...
 
 export interface BasePropsWithAnalyticsMetadata {
+  analyticsMetadata?: AnalyticsMetadata;
   __analyticsMetadata?: AnalyticsMetadata;
 }
 
 export function getAnalyticsMetadataProps(props?: unknown) {
-  return (props as BasePropsWithAnalyticsMetadata | undefined)?.__analyticsMetadata;
+  const propsWithAnalytics = props as any as BasePropsWithAnalyticsMetadata;
+  return propsWithAnalytics?.analyticsMetadata || propsWithAnalytics?.__analyticsMetadata || {};
```
Comment (michaeldowseza) [inline src/internal/base-component/index.ts:48]: Backwards compatibility 🤷🏻‍♂️

File: src/__tests__/__snapshots__/documenter.test.ts.snap [hunk 2]
```diff
@@ -7366,6 +7417,7 @@ use the \`id\` attribute, consider setting it on a parent element instead.",
       "description": "Specifies flash messages that appear in the same order that they are listed.
 The value is an array of flash message definition objects.
 A flash message object contains the following properties:
+* \`analyticsMetadata\` (AnalyticsMetadata) - (Optional) Specifies additional analytics related metadata.
```
Comment (connorlanigan) [inline src/__tests__/__snapshots__/documenter.test.ts.snap:7420]: (minor) i'd recommend to hyphenate: "analytics-related"

File: src/__tests__/__snapshots__/documenter.test.ts.snap [hunk 3]
```diff
@@ -7411,6 +7462,28 @@ Object {
   "functions": Array [],
   "name": "Form",
   "properties": Array [
+    Object {
+      "description": "",
+      "inlineType": Object {
+        "name": "AnalyticsMetadata",
```
Comment (connorlanigan) [inline src/__tests__/__snapshots__/documenter.test.ts.snap:7468]: In the other components, the type was prefixed with the component name, but here for Form it is jus `AnalyticsMetadata` (same in line 16662). Is this intended? Could it cause problems? We could probably prevent it by moving these into a `Form` namespace

File: src/internal/base-component/index.ts [hunk 4]
```diff
@@ -39,9 +39,11 @@ export function getBaseProps(props: BaseComponentProps) {
 }
 
 export interface BasePropsWithAnalyticsMetadata {
+  analyticsMetadata?: AnalyticsMetadata;
   __analyticsMetadata?: AnalyticsMetadata;
 }
 
 export function getAnalyticsMetadataProps(props?: unknown) {
-  return (props as BasePropsWithAnalyticsMetadata | undefined)?.__analyticsMetadata;
+  const propsWithAnalytics = props as any as BasePropsWithAnalyticsMetadata;
```
Comment (connorlanigan) [inline src/internal/base-component/index.ts:47]: Do we still need the unsafe type assertion? Now that each component has the `analyticsMetadata` property in its public types, we could replace `unknown` in the parameter with something more strict like `BasePropsWithAnalyticsMetadata`

File: src/__tests__/__snapshots__/documenter.test.ts.snap (hunk 3 above)
Comment (michaeldowseza) [reply src/__tests__/__snapshots__/documenter.test.ts.snap:7468]: Moved

File: src/internal/base-component/index.ts (hunk 4 above)
Comment (michaeldowseza) [reply src/internal/base-component/index.ts:47]: Changed

File: src/__tests__/__snapshots__/documenter.test.ts.snap (hunk 2 above)
Comment (michaeldowseza) [reply src/__tests__/__snapshots__/documenter.test.ts.snap:7420]: Changed.

**Review by connorlanigan — APPROVED** [review]

**Review by connorlanigan — DISMISSED** [review]

**Review by connorlanigan — APPROVED** [review]

**Review by connorlanigan — APPROVED** [review]

**Review by connorlanigan — DISMISSED** [review]

**Review by connorlanigan — APPROVED** [review]

**Pull request merged**